packages/react-aria-components/docs/DropZone.mdx

```diff
@@ -0,0 +1,362 @@
+The `className` and `style` props also accept functions which receive states for styling. This lets you dynamically determine the classes or styles to apply, which is useful when using utility CSS libraries like [Tailwind](https://tailwindcss.com/).
+
+```jsx
+<DropZone className={({isDropTarget}) => isDropTarget ? 'bg-gray-700' : 'bg-gray-600'} />
+```
+
+Render props may also be used as children to alter what elements are rendered based on the current state. For example, you could render an extra element when the button is in a pressed state.
+
+```jsx
+<DropZone>
+  {({isDropTarget}) => (
+    <>
+      {isDropTarget && <DropHighlight/>}
+      Drop item here
+    </>
+  )}
+</DropZone>
+```
+
+The states, selectors, and render props for `DropZone` are documented below.
+
+<StateTable properties={docs.exports.DropZoneRenderProps.properties} />
+
+## Usage
+
+### Events
+
+`DropZone` supports user interactions via mouse, keyboard, and touch. You can handle all of these via the `onDrop` prop. In addition, the `onDropEnter`, `onDropMove`, and `onDropExit` events are fired as the user interacts with the dropzone. 
+
+```tsx example
+import type {TextDropItem} from '@react-aria/dnd';
+
+function Example() {
+  let [dropped, setDropped] = React.useState(null);
+
+  return (
+    <>
+      <Draggable />
+      <DropZone
+        onDrop={async(e) => {
+          let items = await Promise.all(
+            e.items
+              .filter(item => item.kind === 'text' && item.types.has('text/plain'))
+              .map((item: TextDropItem) => item.getText('text/plain'))
+          );
+          setDropped(items.join('\n'));
+        }}>
+        <Text slot="heading">
+          {dropped || 'Drop here'}
+        </Text>
+      </ DropZone>
+    </>
+  );
+}
+
+<Example />
+
+```
+
+#### Draggable
+
+The `Draggable` component used above is defined below. See [useDrag](useDrag.html) for more details and documentation.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example render=false export=true
+import {useDrag} from '@react-aria/dnd';
+
+function Draggable() {
+  let {dragProps, isDragging} = useDrag({
+    getItems() {
+      return [{
+        'text/plain': 'hello world',
+        'my-app-custom-type': JSON.stringify({message: 'hello world'})
+      }];
+    }
+  });
+
+  return (
+    <div {...dragProps} role="button" tabIndex={0} className={`draggable ${isDragging ? 'dragging' : ''}`}>
+      Drag me
+    </div>
+  );
+}
+```
+</details>
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.draggable {
+  display: inline-block;
+  vertical-align: top;
+  border: 1px solid gray;
+  padding: 10px;
+  margin-right: 20px;
+  border-radius: 4px;
+}
+
+.draggable.dragging {
+  opacity: 0.5;
+}
+```
+
+</details>
+
+### Heading
+
+The `heading` slot enables the user to reference the text content to define the dropzone's accessible name. 
```

Comment:
Seeing this here I think we should probably rename the `heading` slot to `label` since it isn't on a `<Heading>` element. We seem to use "label" in a bunch of other components already.